InteroperabilityTest_Engine/Compute/Compare.cs

```diff
@@ -0,0 +1,391 @@
+        [Input("refResult", "Reference result to compare to.")]
+        [Output("result", "A test result containing information about the equality of the two input TestResults.")]
+        public static TestResult Compare(TestResult result, TestResult refResult)
+        {
+            if (result == null)
+                return null;
+
+            if (refResult == null)
+                return result.NoReferenceFound();
+
+            TestResult diffResult = new TestResult() { Description = $"PushPullCompare difference: {result.ID}", ID = result.ID };
+
+            //Check if same type, i.e. Pass/Warning/Error
+            if (result.Status == refResult.Status)
+            {
+                if (result.Status == oM.Test.TestStatus.Warning)
+                {
+                    //If of type Warning, check the inner results
+                    List<oM.Test.ITestInformation> referenceInformation = refResult.NonEventMessageInformation();
+                    List<oM.Test.ITestInformation> referenceInformationNoMatch = referenceInformation.ToList();
+
+                    foreach (oM.Test.ITestInformation innerInformation in result.NonEventMessageInformation())
+                    {
+                        //Find matching reference data
+                        oM.Test.ITestInformation reference = referenceInformation.IGetSameIdInformation(innerInformation);
+
+                        //Compare the inner test information
+                        oM.Test.ITestInformation comparisonResult = ICompare(innerInformation, reference);
+                        if (comparisonResult != null)
+                            diffResult.Information.Add(comparisonResult);
+
+                        if (reference != null)
+                            referenceInformationNoMatch.Remove(reference);
+                    }
+
+                    foreach (oM.Test.ITestInformation refInfo in referenceInformationNoMatch)
+                    {
+                        diffResult.Information.Add(refInfo.IOnlyReferenceFound());
+                    }
+                }
+                else
+                {
+                    //Else, the results are the same
+                    diffResult.Message = "No change in the result status. Was and still is: " + result.Status + ". No changes in differences found from reference data.";
+                    diffResult.Status = oM.Test.TestStatus.Pass;
+                    return diffResult;
+                }
+            }
+            //Result Status are different
+            else
+            {
+                //Reference is pass, current is not
+                if (refResult.Status == oM.Test.TestStatus.Pass)
+                {
+                    if (result.Status == oM.Test.TestStatus.Error)
+                    {
+                        //System is now crashing, was working before
+                        diffResult.Message = "An error has been introduced that was previously fully passing!";
```

Comment:
```suggestion
                        diffResult.Message = "An error has been introduced that did not exist previously.";
```

I would tend to avoid the use of `!` on the messages - these will be reported back by bots and can make things seem more 'excitable' than they need to be.